app/services/utils/collection.py

```diff
@@ -0,0 +1,56 @@
+from typing import Optional, Dict, Any
+
+import requests
+
+
+def get_collection_items_url(
+    base_url: str, collection_id: str
+) -> Optional[str]:
+    url = base_url + collection_id
+    response = requests.get(url)
+    response.raise_for_status()
```

Comment:
Instead of just raising the error verify if it's 404 and return a user friendly message (the same as you had before)